shell/platform/darwin/ios/platform_view_ios.h

```diff
@@ -50,6 +50,7 @@ class PlatformViewIOS final : public PlatformView {
  private:
   fml::WeakPtr<FlutterViewController> owner_controller_;
   std::unique_ptr<IOSSurface> ios_surface_;
+  std::mutex ios_surface_mutex_;
```

Comment:
You can annotate ios_surface_ with [FML_GUARDED_BY](https://github.com/flutter/engine/blob/master/fml/synchronization/thread_annotations.h) so that static thread safety check can be enabled. Also, can you add a comment there saying which threads access that resource and when. Since there wont be any contention, its hard to reason about why that lock is necessary.

Reply:
Done.

I'm not so sure usage of ios_surface_ is coordinated externally, calling `SetOwnerViewController` rapidly could cause `CreateRenderingSurface` to be talking to a undefined ios_surface_ without this mutex.

Reply:
That good to document too. But your call.